src/optionals/math.c

```diff
@@ -221,6 +221,92 @@ static Value tanNative(DictuVM *vm, int argCount, Value *args) {
     return NUMBER_VAL(tan(AS_NUMBER(args[0])));
 }
 
+static long long gcd(long long a, long long b) {
+    long long r;
+    while (b > 0) {
+        r = a % b;
+        a = b;
+        b = r;
+    }
+    return a;
+}
+
+static Value gcdNative(DictuVM *vm, int argCount, Value *args) {
+    if (argCount == 1 && IS_LIST(args[0])) {
+        ObjList *list = AS_LIST(args[0]);
+        argCount = list->values.count;
+        args = list->values.values;
+    }
+
+    if (argCount < 2) {
+        runtimeError(vm, "gcd() requires 2 or more argument (%d given).", argCount);
+        return EMPTY_VAL;
+    }
+
+    for (int i = 0; i < argCount; ++i)
+        if (!IS_NUMBER(args[i])) {
+            runtimeError(vm, "A non-number value passed to gcd() as argument #%d", i);
```

Comment:
These will be 0-based. From a UX perspective for the error, I think 1-based would be a better solution. Do you agree?

Reply:
I thought about that, and was unsure. Since devs are the end-user, maybe they'd think "argument #1" is the second argument? I'm currently trying out a new way that should be less ambiguous, and make the error messages clear if a list is given.